tests/client/test_transaction_marks.py

```diff
@@ -0,0 +1,70 @@
+import time
+
+import pytest
+
+import elasticapm
+
+
+def test_basic_mark(elasticapm_client):
+    elasticapm_client.begin_transaction('test')
+    elasticapm.mark('test_group', 'test')
+    elasticapm_client.end_transaction('test')
+    transaction = elasticapm_client.instrumentation_store.get_all()[0]
+    assert transaction['marks']['test_group']['test'] > 0
+
+
+def test_basic_multiple_mark_with_override(elasticapm_client):
+    elasticapm_client.begin_transaction('test')
+    elasticapm.mark('test_group', 'test')
+    time.sleep(0.1)
+    elasticapm.mark('test_group', 'test')
+    elasticapm_client.end_transaction('test')
+    transaction = elasticapm_client.instrumentation_store.get_all()[0]
+    assert transaction['marks']['test_group']['test'] > 0.1
+    assert len(transaction['marks']['test_group']) == 1
+
+
+def test_basic_multiple_mark_without_override(elasticapm_client):
+    elasticapm_client.begin_transaction('test')
+    for i in range(5):
+        elasticapm.mark('test_group', 'test', override=False)
+    elasticapm_client.end_transaction('test')
+    transaction = elasticapm_client.instrumentation_store.get_all()[0]
+    assert transaction['marks']['test_group']['test'] > 0
+    for i in range(1, 5):
+        assert transaction['marks']['test_group']['test-%d' % i] > 0
+
+
+def test_mark_while_no_transaction(elasticapm_client, caplog):
+    elasticapm.mark('test_group', 'test')
+    assert len(caplog.records) == 1
+    assert "No transaction currently active" in caplog.records[0].message
+
+
+@pytest.mark.parametrize('elasticapm_client', [
+    {'transaction_mark_errors': 'all'},
+    {'transaction_mark_errors': 'handled'},
+    {'transaction_mark_errors': 'unhandled'},
+    {'transaction_mark_errors': 'off'},
+], indirect=True)
+def test_mark_errors(elasticapm_client):
```

Comment:
I think in this case you might be better off creating separate test methods instead of using if statements inside the method. You could extract a method for the begin/end transaction to get rid of the repetition.